### PR TITLE
Add support for http based repositories

### PR DESCRIPTION
**Pull request by GRMrGecko**
Hello,

I had a need to add http repository support to this project for a project I was working on. As this is open source, I am contributing back. Hope this gets added, as I believe it is useful.

## Summary by CodeRabbit

- **New Features**
	- Introduced support for HTTP-based repositories, enabling software updates from custom HTTP sources.
	- Added a new section in the documentation detailing the `HttpSource` functionality and providing usage examples.

- **Bug Fixes**
	- Improved clarity and flow of the README documentation.

- **Tests**
	- Implemented a comprehensive suite of unit tests for HTTP source functionalities to ensure reliability and correct behavior.

- **Chores**
	- Added a new YAML configuration file for testing release information, including multiple release entries and associated assets.

### CODE REVIEW

File: http_release.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import (
+	"time"
+)
+
+type HttpAsset struct {
+	ID   int64  `yaml:"id"`
+	Name string `yaml:"name"`
+	Size int    `yaml:"size"`
+	URL  string `yaml:"url"`
+}
```
[the remark this answers is not in the document]
Comment (GRMrGecko) [reply http_release.go:32]: This is normal, the size is the file size and the ID is the ID of the asset.

File: http_source.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"gopkg.in/yaml.v3"
+)
```
[the remark this answers is not in the document]
Comment (GRMrGecko) [reply http_source.go]: Not necessarily wrong as is, but it doesn't hurt I guess...

File: http_release.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	ReleaseID    int64        `yaml:"release_id"`
+	Name         string       `yaml:"name"`
+	TagName      string       `yaml:"tag_name"`
+	URL          string       `yaml:"url"`
+	Draft        bool         `yaml:"draft"`
+	Prerelease   bool         `yaml:"prerelease"`
+	PublishedAt  time.Time    `yaml:"published_at"`
+	ReleaseNotes string       `yaml:"release_notes"`
+	Assets       []*HttpAsset `yaml:"assets"`
+}
```
[the remark this answers is not in the document]
Comment (GRMrGecko) [reply http_release.go:62]: Go format aligned it this way.

[the remark this answers is not in the document]
Comment (GRMrGecko) [reply http_release.go:62]: Oh, didn't fully read what was said. The reason behind the release ID field name is to match what other sources in the project uses for field names. I could name it id, but then I'd have to update code in other projects. If the maintainer wants to do that, I can update.

[the remark this answers is not in the document]
Comment (GRMrGecko) [reply http_release.go]: Gone ahead and made the change after thinking about it more, going to mark this as resolved.

**Comment by GRMrGecko** [comment]
Hopefully last commit to cover your AI requests.

**Comment by creativeprojects** [comment]
Well, that makes a nice addition indeed 👍🏻 

I'm merging it now, then I'm going to try a few things with it before releasing a new version 😉 

Thank you very much 🙇🏻

**Pull request merged**